File: CSharp/Mayope.Api/Requests/ListBranchesRequest.cs
using System;
using System.ComponentModel.DataAnnotations;

using Newtonsoft.Json;

namespace Mayope.Api.Requests;
{
    /// <summary>
    /// A request for a list of branches.
    /// </summary>
    public class ListBranchesRequest : AuthenticatedRequest
    {
        /// <summary>
        /// The UUID of the tree to list the branches in.
        /// </summary>
        [Required]
        [JsonRequired]
        public Guid Tree { get; set; }
    }
}

File: Java/org/mayope/api/request/ListBranchesRequest.java
package org.mayope.api.request;

import java.util;

/**
 * A request for a list of branches.
 */
public class ListBranchesRequest extends AuthenticatedRequest {
    private UUID tree;

    public UUID getTree() {
        return tree;
    }

    public void setTree(UUID tree) {
        this.tree = tree;
    }
}
